### PR TITLE
[management] Add version gate to stop sending deprecated RemotePeers field

### PR DESCRIPTION
**Pull request by mlsmaycon**
## Describe your changes
Don't send top-level remote peers to peers in the  v0.29.3 or newer

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [x] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] This change does **not** modify the public API, gRPC protocols, functionality behavior, CLI / service flags, or introduce a new feature — **OR** I have discussed it with the NetBird team beforehand (link the issue / Slack thread in the description). See [CONTRIBUTING.md](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTING.md#discuss-changes-with-the-netbird-team-first).

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **Chores**
  * Peer sync now omits legacy peer details for clients on deprecated or development builds and exposes peers via a nested network map, reducing legacy noise and improving consistency.

* **Tests**
  * Added table-driven tests covering version formats, dev-build variants, and boundary cases for the compatibility-gating behavior.
  * Updated sync tests to validate peers via the nested network map.

### CODE REVIEW

**Review by pascal-fischer — DISMISSED** [review]

**Review by pappz — APPROVED** [review]

File: management/internals/shared/grpc/conversion.go [hunk 1]
```diff
@@ -246,6 +252,33 @@ func buildAuthorizedUsersProto(ctx context.Context, authorizedUsers map[string]m
 	return hashedUsers, machineUsers
 }
 
+const deprecatedRemotePeersVersion = "0.29.3"
```
Comment (pappz) [inline management/internals/shared/grpc/conversion.go]: Please move the const, vars init() to the beginning of the file.

**Review by pappz — CHANGES_REQUESTED** [review]

File: management/internals/shared/grpc/conversion.go (hunk 1 above)
Comment (mlsmaycon) [reply management/internals/shared/grpc/conversion.go]: pushed

**Review by pappz — DISMISSED** [review]

**Review by pappz — APPROVED** [review]
